hack/sdn-kind-cno.sh

```diff
@@ -0,0 +1,239 @@
+
+if [ "$BUILD_SDN" = true ] || [ "$BUILD_CNO" = true ]; then
+  pushd $CNO_TEMPLATES
+  DEPLOYMENT_TEMPLATE=$(ls 0000*deployment*)
+  if [ -z "$DEPLOYMENT_TEMPLATE" ]; then
+    echo "error locating deployment template in $CNO_TEMPLATES"
+    exit 1
+  fi
+  cp $DEPLOYMENT_TEMPLATE deployment.yaml.bk
+  if [ "$BUILD_SDN" = true ]; then
+    sed -i 's#quay.io/openshift/origin-sdn:latest#sdn-test#' $DEPLOYMENT_TEMPLATE
+  fi
+  if [ "$BUILD_CNO" = true ]; then
+    sed -i "s#quay.io/openshift/origin-cluster-network-operator:.*#$CNO_IMAGE#" $DEPLOYMENT_TEMPLATE
+  fi
+fi
+
+echo "Creating \"cluster-config-v1\" configMap with $NUM_MASTER_NODES master nodes"
+cat <<EOF | kubectl create -f - 
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-config-v1
+  namespace: kube-system
+data:
+  install-config: |
+    apiVersion: v1
+    controlPlane:
+      replicas: ${NUM_MASTER_NODES}
+EOF
+
+echo "Creating SDN CNO config"
+cat << EOF | kubectl create -f -
+apiVersion: config.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  clusterNetwork:
+  - cidr: ${CLUSTER_CIDR}
+    hostPrefix: ${HOST_PREFIX}
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - ${SERVICE_NETWORK}
+EOF
+
+echo "Creating CNO operator"
+for f in $(ls $CNO_TEMPLATES| grep 0000| grep -v credentials); do
+  kubectl create -f ${CNO_TEMPLATES}/$f
+done
+
+# OVS is expected to run in systemd but that not an option in kindest/noede
+# we need to deploy it in a pod.
+echo "Creating OVS daemonset"
+kubectl create namespace ovs-kind
+kubectl create -f $CNO_PATH/hack/ovs-kind.yaml
+
+if [ "$BUILD_SDN" = true ] || [ "$BUILD_CNO" = true ]; then
+  mv deployment.yaml.bk $DEPLOYMENT_TEMPLATE
+  popd
+fi
+
+if ! kubectl wait -n openshift-network-operator --for condition=available deployment network-operator --timeout=120s; then
+  echo "Network operator not running"
+  exit 1
+fi
+
+
+
+echo "Deleting default kube-proxy"
+kubectl delete -n kube-system ds kube-proxy
+
+if [ "$BUILD_CNO" != true ]; then
+  echo "WARNING: patching CNO operator pod for OVN-K8S, deployment will no longer function if this pod is restarted"
+  CNO_POD=$(kubectl get pod -n openshift-network-operator -o jsonpath='{.items[0].metadata.name}' --field-selector status.phase=Running)
+  # kubectl -n openshift-network-operator  exec $CNO_POD sed -i '/host-run-netns/{n;s/readOnly.*/mountPropagation: Bidirectional/}' /bindata/network/ovn-kubernetes/ovnkube-node.yaml > /tmp/ovnkube-node.yaml
+  # kubectl cp /tmp/ovnkube-node.yaml openshift-network-operator/${CNO_POD}:/bindata/network/ovn-kubernetes/
+fi
+
+
+for n in $NODES; do
+  echo "Sym-linking cni dirs for node $n"
+  docker exec $n rm -rf /opt/cni/bin
+  docker exec $n ln -s /var/lib/cni/bin /opt/cni/
+  docker exec $n rm -rf /etc/cni/net.d
+  docker exec $n mkdir -p /etc/cni/
+  docker exec $n ln -s /etc/kubernetes/cni/net.d /etc/cni/
+done
+
+# wait until resources are created.
+sleep 150
```

Comment:
this sleep should not be needed since we have an active loop later with `kubectl wait`

Reply:
kubectl wait --for=condition=ready fails if there aren't any pods at all with that selector.
I added the sleep afterwards, anyway 150 isn't long enough so I'd like to keep it as is.